mtenn/config.py

```diff
@@ -577,14 +578,16 @@ def massage_into_lists(cls, values) -> GATModelConfig:
         # If we just want a model with one layer, can return early since we've already
         #  converted everything into lists
         if num_layers == 1:
-            return values
+            # update self with the new values
+            self.__dict__.update(values)
+            
 
         # Adjust any length 1 list to be the right length
         for p, list_len in list_lens.items():
             if list_len == 1:
                 values[p] = values[p] * num_layers
 
-        return values
+        return self.__dict__.update(values)
```

Comment:
I don't think `update` actually returns anything
```suggestion
        self.__dict__.update(values)
        return self.__dict__
```

Reply:
Now that you mention it we should return `self`

Reply:
this is due to how the `after` mode validator works in pydantic 2.